futures-util/src/async_await/random.rs

```diff
@@ -52,3 +53,26 @@ fn random() -> u64 {
         x.0.wrapping_mul(0x2545_f491_4f6c_dd1d)
     })
 }
+
+#[cfg(not(feature = "std"))]
+fn random() -> u64 {
+    use core::sync::atomic::{AtomicUsize, Ordering};
+
+    static RNG: AtomicUsize = AtomicUsize::new(1);
+
+    let mut x = RNG.load(Ordering::Relaxed);
+
+    if core::mem::size_of::<usize>() == 4 {
+        x ^= x << 13;
+        x ^= x >> 17;
+        x ^= x << 5;
+    } else if core::mem::size_of::<usize>() == 8 {
+        x ^= x >> 12;
+        x ^= x << 25;
+        x ^= x >> 27;
+        x = x.wrapping_mul(0x2545_f491_4f6c_dd1d);
```

Comment:
This looks like repeated code, would be nice if we could refactor this somehow.

Reply:
Thank you for your review.
I prepared `xorshift64star` and `xorshift32` functions.

Reply:
@ytakano thanks, I should have mentioned that I am just an outside contributor, hope we can get a review from @taiki-e